symphony-bdk-spring/symphony-bdk-core-spring-boot-starter/src/main/java/com/symphony/bdk/spring/annotation/SlashAnnotationProcessor.java

```diff
@@ -0,0 +1,155 @@
+            if (targetClass != null) {
+              type = targetClass;
+            }
+          }
+          catch (Throwable ex) {
+            // An invalid scoped proxy arrangement - let's ignore it.
+            log.debug("Could not resolve target bean for scoped proxy '" + beanName + "'", ex);
+          }
+        }
+        try {
+          this.processBean(bean, beanName, type);
+        }
+        catch (Throwable ex) {
+          throw new BeanInitializationException(
+              "Failed to process @Slash annotation on bean with name '" + beanName + "'", ex);
+        }
+      }
+    }
+
+    return bean;
+  }
+
+  private void processBean(final Object bean, final String beanName, final Class<?> targetType) {
+    if (AnnotationUtils.isCandidateClass(targetType, Slash.class)
+        && !isSpringContainerClass(targetType)
+        && !isBdkCoreClass(targetType)) {
+
+      Map<Method, Slash> annotatedMethods = null;
+
+      try {
+        annotatedMethods = MethodIntrospector.selectMethods(targetType,
+            (MethodIntrospector.MetadataLookup<Slash>) method ->
+                AnnotatedElementUtils.findMergedAnnotation(method, Slash.class));
+      }
+      catch (Throwable ex) {
+        // An unresolvable type in a method signature, probably from a lazy bean - let's ignore it.
+        log.debug("Could not resolve methods for bean with name '" + beanName + "'", ex);
+      }
+
+      if (!CollectionUtils.isEmpty(annotatedMethods)) {
+        for (final Method m : annotatedMethods.keySet()) {
+
+          final Slash annotation = AnnotationUtils.getAnnotation(m, Slash.class);
+
+          if (annotation != null) {
+            if (isMethodPrototypeValid(m)) {
+              this.registerSlashMethod(bean, m, annotation);
+            } else {
+              log.warn("Method '{}' is annotated by @Slash but does not respect the expected prototype. "
+                  + "It must accept a single argument of type '{}'", m, CommandContext.class);
+            }
+          }
+        }
+      }
+    }
+  }
+
+  private void registerSlashMethod(Object bean, Method method, Slash annotation) {
+    this.registry.register(slash(annotation.value(), annotation.mentionBot(), c -> {
+      try {
+        method.invoke(bean, c);
+      } catch (IllegalAccessException | InvocationTargetException e) {
+        log.error("Unable to invoke @Slash method {} from bean {}", method.getName(), bean.getClass(), e);
+      }
+    }));
+  }
+
+  private static boolean isMethodPrototypeValid(Method m) {
+    return m.getParameterCount() == 1 && m.getParameters()[0].getType().equals(CommandContext.class);
+  }
+
+  /**
+   * Determine whether the given class is an {@code org.springframework}
+   * bean class that is not annotated as a user or test {@link Component}...
+   * which indicates that there is no {@link Slash} to be found there.
+   */
+  private static boolean isSpringContainerClass(Class<?> clazz) {
```

Comment:
is that even possible to have a spring class annotated with @ Slash ?

Reply:
Called as `!isSpringContainerClass(...)` earlier in the class. Maybe I should rename and invert to `isNotSpringContainerClass(...)`